### PR TITLE
adding example for running from branch

### PR DESCRIPTION
**Pull request by esc**
Adds an example of how to run the integration-testing from a branch to
the README. This might come in handy when doing this again in future.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -49,6 +49,55 @@ In order to add a new target, you need to:
   tested.
 * Submit a pull-request on Github.
 
+Testing a Branch or Pull-Request
+================================
+
+Normally, Numba is installed as a conda packge from https://anaconda.org using
+a ``CondaSource`` configuration. Sometimes it can be useful to run the
+integration-testing from a branch or a pull-request however. The following
```
Comment (stuartarchibald) [inline README.rst]: Perhaps put `However, ` at the start of the sentence and not at the end?

File: README.rst [hunk 2]
```diff
@@ -49,6 +49,55 @@ In order to add a new target, you need to:
...
+configuration demonstrates how to obtain the branch ``refactor_it``
+from the Github fork at ``github.com/esc/numba``.
+
+Please be advised that you must mirror the tags of the blessed Numba
```
Comment (stuartarchibald) [inline README.rst]: "blessed", whilst I know what you mean, I think `upstream` might be better?

File: README.rst [hunk 3]
```diff
@@ -49,6 +49,55 @@ In order to add a new target, you need to:
...
+repository at ``github.com/numba/numba`` to the desired fork in such cases.
+This is because the Numba version is determined from the closest reachable
+tag in the Git history so recent tags must be present for the build system
+to accurately determine the Numba version. Otherwise you may end up with a
+nonsensical version number that is likely to confuse.
```
Comment (stuartarchibald) [inline README.rst]:
```suggestion
nonsensical version number that is likely to cause confusion.
```

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch. Few minor suggestions, else looks good.

**Comment by esc** [comment]
@stuartarchibald thanks for the review, changes have been addressed.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the fixes!

**Comment by stuartarchibald** [comment]
@esc feel free to merge this as convenient.

**Pull request merged**